rust/pact_consumer/src/mock_server.rs

```diff
@@ -32,26 +51,53 @@ impl StartMockServer for Pact {
 pub struct ValidatingMockServer {
     // A description of our mock server, for use in error messages.
     description: String,
-    // The ID of our mock server.
-    port: i32,
     // The URL of our mock server.
     url: Url,
+    // The mock server instance
+    mock_server: mock_server::MockServer,
+    // The running mode of our mock server.
+    _mode: Mode
 }
 
 impl ValidatingMockServer {
+    /// Create a new mock server
```

Comment:
I think there should be tests here for both of the running modes of the validating mock server, and perhaps better documentation overall.

The reason there is two running modes is as you may guess the ability to test both synchronous and asynchronous http clients. When http clients under test are async, tests are more performant when running the mock server without allocating any extra threads.